Tuist/Config.swift

```diff
@@ -7,4 +7,6 @@ let config = Config(
         options: [.optional]
     ),
     swiftVersion: .init("5.8")
+    // TODO: Enable after https://github.com/tuist/tuist/pull/5632 is merged
+    // generationOptions: .options(staticSideEffectsWarningTargets: .none)
```

Comment:
My advise for this warning (even for tuist) wouldn't to outright dismiss it. 

At the very list enumerated the handful of targets flagging the warning after reviewing if indeed it is ok and that way we'd still get a warning for new issues to re-review rather than them getting hidden.

Having examined Tuist, the issue is being flagged with `TuistAcceptanceTesting`:


```
 · Target 'AEXML' has been linked from target 'TuistAcceptanceTesting' and target 'TuistBuildAcceptanceTests', it is a static product so may introduce unwanted side effects.
 · Target 'AEXML' has been linked from target 'TuistAcceptanceTesting' and target 'TuistGenerateOneAcceptanceTests', it is a static product so may introduce unwanted side effects.
 · Target 'AEXML' has been linked from target 'TuistAcceptanceTesting' and target 'TuistTestAcceptanceTests', it is a static product so may introduce unwanted side effects.
 · Target 'AnyCodable' has been linked from target 'TuistAcceptanceTesting' and target 'TuistBuildAcceptanceTests', it is a static product so may introduce unwanted side effects.
 · Target 'AnyCodable' has been linked from target 'TuistAcceptanceTesting' and target 'TuistGenerateOneAcceptanceTests', it is a static product so may introduce unwanted side effects.
 · Target 'AnyCodable' has been linked from target 'TuistAcceptanceTesting' and target 'TuistTestAcceptanceTests', it is a static product so may introduce unwanted side effects.
 · Target 'CSystem' has been linked from target 'TuistAcceptanceTesting' and target 'TuistBuildAcceptanceTests', it is a static product so may introduce unwanted side effects.
 · Target 'CSystem' has been linked from target 'TuistAcceptanceTesting' and target 'TuistGenerateOneAcceptanceTests', it is a static product so may introduce unwanted side effects.
 · Target 'CSystem' has been linked from target 'TuistAcceptanceTesting' and target 'TuistTestAcceptanceTests', it is a static product so may introduce unwanted side effects.
 · Target 'CombineExt' has been linked from target 'TuistAcceptanceTesting' and target 'TuistBuildAcceptanceTests', it is a static product so may introduce unwanted side effects.
 · Target 'CombineExt' has been linked from target 'TuistAcceptanceTesting' and target 'TuistGenerateOneAcceptanceTests', it is a static product so may introduce unwanted side effects.
 · Target 'CombineExt' has been linked from target 'TuistAcceptanceTesting' and target 'TuistTestAcceptanceTests', it is a static product so may introduce unwanted side effects.
 · Target 'KeychainAccess' has been linked from target 'TuistAcceptanceTesting' and target 'TuistBuildAcceptanceTests', it is a static product so may introduce unwanted side effects.
 · Target 'KeychainAccess' has been linked from target 'TuistAcceptanceTesting' and target 'TuistGenerateOneAcceptanceTests', it is a static product so may introduce unwanted side effects.
 · Target 'KeychainAccess' has been linked from target 'TuistAcceptanceTesting' and target 'TuistTestAcceptanceTests', it is a static product so may introduce unwanted side effects.
 · Target 'Logging' has been linked from target 'TuistAcceptanceTesting' and target 'TuistBuildAcceptanceTests', it is a static product so may introduce unwanted side effects.
 · Target 'Logging' has been linked from target 'TuistAcceptanceTesting' and target 'TuistGenerateOneAcceptanceTests', it is a static product so may introduce unwanted side effects.
 · Target 'Logging' has been linked from target 'TuistAcceptanceTesting' and target 'TuistTestAcceptanceTests', it is a static product so may introduce unwanted side effects.
 · Target 'PathKit' has been linked from target 'TuistAcceptanceTesting' and target 'TuistBuildAcceptanceTests', it is a static product so may introduce unwanted side effects.
 · Target 'PathKit' has been linked from target 'TuistAcceptanceTesting' and target 'TuistGenerateOneAcceptanceTests', it is a static product so may introduce unwanted side effects.
 · Target 'PathKit' has been linked from target 'TuistAcceptanceTesting' and target 'TuistTestAcceptanceTests', it is a static product so may introduce unwanted side effects.
 · Target 'Swifter' has been linked from target 'TuistAcceptanceTesting' and target 'TuistBuildAcceptanceTests', it is a static product so may introduce unwanted side effects.
 · Target 'Swifter' has been linked from target 'TuistAcceptanceTesting' and target 'TuistGenerateOneAcceptanceTests', it is a static product so may introduce unwanted side effects.
 · Target 'Swifter' has been linked from target 'TuistAcceptanceTesting' and target 'TuistTestAcceptanceTests', it is a static product so may introduce unwanted side effects.
 · Target 'SystemPackage' has been linked from target 'TuistAcceptanceTesting' and target 'TuistBuildAcceptanceTests', it is a static product so may introduce unwanted side effects.
 · Target 'SystemPackage' has been linked from target 'TuistAcceptanceTesting' and target 'TuistGenerateOneAcceptanceTests', it is a static product so may introduce unwanted side effects.
 · Target 'SystemPackage' has been linked from target 'TuistAcceptanceTesting' and target 'TuistTestAcceptanceTests', it is a static product so may introduce unwanted side effects.
 · Target 'TSCBasic' has been linked from target 'TuistAcceptanceTesting' and target 'TuistBuildAcceptanceTests', it is a static product so may introduce unwanted side effects.
 · Target 'TSCBasic' has been linked from target 'TuistAcceptanceTesting' and target 'TuistGenerateOneAcceptanceTests', it is a static product so may introduce unwanted side effects.
 · Target 'TSCBasic' has been linked from target 'TuistAcceptanceTesting' and target 'TuistTestAcceptanceTests', it is a static product so may introduce unwanted side effects.
 · Target 'TSCLibc' has been linked from target 'TuistAcceptanceTesting' and target 'TuistBuildAcceptanceTests', it is a static product so may introduce unwanted side effects.
 · Target 'TSCLibc' has been linked from target 'TuistAcceptanceTesting' and target 'TuistGenerateOneAcceptanceTests', it is a static product so may introduce unwanted side effects.
 · Target 'TSCLibc' has been linked from target 'TuistAcceptanceTesting' and target 'TuistTestAcceptanceTests', it is a static product so may introduce unwanted side effects.
 · Target 'TSCUtility' has been linked from target 'TuistAcceptanceTesting' and target 'TuistBuildAcceptanceTests', it is a static product so may introduce unwanted side effects.
 · Target 'TSCUtility' has been linked from target 'TuistAcceptanceTesting' and target 'TuistGenerateOneAcceptanceTests', it is a static product so may introduce unwanted side effects.
 · Target 'TSCUtility' has been linked from target 'TuistAcceptanceTesting' and target 'TuistTestAcceptanceTests', it is a static product so may introduce unwanted side effects.
 · Target 'TSCclibc' has been linked from target 'TuistAcceptanceTesting' and target 'TuistBuildAcceptanceTests', it is a static product so may introduce unwanted side effects.
 · Target 'TSCclibc' has been linked from target 'TuistAcceptanceTesting' and target 'TuistGenerateOneAcceptanceTests', it is a static product so may introduce unwanted side effects.
 · Target 'TSCclibc' has been linked from target 'TuistAcceptanceTesting' and target 'TuistTestAcceptanceTests', it is a static product so may introduce unwanted side effects.
 · Target 'TuistCore' has been linked from target 'TuistAcceptanceTesting' and target 'TuistBuildAcceptanceTests', it is a static product so may introduce unwanted side effects.
 · Target 'TuistCore' has been linked from target 'TuistAcceptanceTesting' and target 'TuistGenerateOneAcceptanceTests', it is a static product so may introduce unwanted side effects.
 · Target 'TuistCore' has been linked from target 'TuistAcceptanceTesting' and target 'TuistTestAcceptanceTests', it is a static product so may introduce unwanted side effects.
 · Target 'TuistGraph' has been linked from target 'TuistAcceptanceTesting' and target 'TuistBuildAcceptanceTests', it is a static product so may introduce unwanted side effects.
 · Target 'TuistGraph' has been linked from target 'TuistAcceptanceTesting' and target 'TuistGenerateOneAcceptanceTests', it is a static product so may introduce unwanted side effects.
 · Target 'TuistGraph' has been linked from target 'TuistAcceptanceTesting' and target 'TuistTestAcceptanceTests', it is a static product so may introduce unwanted side effects.
 · Target 'TuistSupport' has been linked from target 'TuistAcceptanceTesting' and target 'TuistBuildAcceptanceTests', it is a static product so may introduce unwanted side effects.
 · Target 'TuistSupport' has been linked from target 'TuistAcceptanceTesting' and target 'TuistGenerateOneAcceptanceTests', it is a static product so may introduce unwanted side effects.
 · Target 'TuistSupport' has been linked from target 'TuistAcceptanceTesting' and target 'TuistTestAcceptanceTests', it is a static product so may introduce unwanted side effects.
 · Target 'TuistSupportTesting' has been linked from target 'TuistAcceptanceTesting' and target 'TuistBuildAcceptanceTests', it is a static product so may introduce unwanted side effects.
 · Target 'TuistSupportTesting' has been linked from target 'TuistAcceptanceTesting' and target 'TuistGenerateOneAcceptanceTests', it is a static product so may introduce unwanted side effects.
 · Target 'TuistSupportTesting' has been linked from target 'TuistAcceptanceTesting' and target 'TuistTestAcceptanceTests', it is a static product so may introduce unwanted side effects.
 · Target 'XcodeProj' has been linked from target 'TuistAcceptanceTesting' and target 'TuistBuildAcceptanceTests', it is a static product so may introduce unwanted side effects.
 · Target 'XcodeProj' has been linked from target 'TuistAcceptanceTesting' and target 'TuistGenerateOneAcceptanceTests', it is a static product so may introduce unwanted side effects.
 · Target 'XcodeProj' has been linked from target 'TuistAcceptanceTesting' and target 'TuistTestAcceptanceTests', it is a static product so may introduce unwanted side effects.
 · Target 'ZIPFoundation' has been linked from target 'TuistAcceptanceTesting' and target 'TuistBuildAcceptanceTests', it is a static product so may introduce unwanted side effects.
 · Target 'ZIPFoundation' has been linked from target 'TuistAcceptanceTesting' and target 'TuistGenerateOneAcceptanceTests', it is a static product so may introduce unwanted side effects.
 · Target 'ZIPFoundation' has been linked from target 'TuistAcceptanceTesting' and target 'TuistTestAcceptanceTests', it is a static product so may introduce unwanted side effects.
```

To fix those without silencing the warning, `TuistAcceptanceTesting` can be changed to be `.staticFramework` - is there a reason it's a dynamic `.framework` ?

Reply:
I changed it to a dynamic framework (`.framework`) and works without any issues.